cyral/internal/repository/confanalysis/resource.go

```diff
@@ -121,6 +121,16 @@ func repositoryConfAnalysisResourceSchemaV0() *schema.Resource {
 				Type:     schema.TypeBool,
 				Optional: true,
 			},
+			"mask_all_occurrences": {
+				Description: "If disabled (default behavior), masking is applicable to all returned values, " +
+					"ensuring users do not get any unmasked data, as defined by policies. (assuming enable_data_masking=true)" +
+					"When enabled, this flag also instructs sidecars that masking should be applied to all " +
+					"occurrences of the sensitive elements covered by masking policies, e.g occurrences in " +
+					"WHERE, HAVING or ON clauses. Enabling this may cause some performance degradation " +
+					"on large tables.",
+				Type:     schema.TypeBool,
+				Optional: true,
```

Comment:
I initially made this change even in the API (via API validations), to validate that mask_all_occurrences and enable_data_masking walk together. My only concern with your proposal is backwards compatibility if someone already has masking enabled. Would a terraform update cause any issues (considering the user wouldn't have mask_all_occurrences define in their terraform file)?

Anyway, I don't mind this change. If you want to include it, we can include it.

Reply:
AFAIK, the `RequiredWith` here would be a dependency from `mask_all_occurrences` to `enable_data_masking`, so it would be backwards compatible as `enable_data_masking` would not depend on `mask_all_occurrences`. Could you please test the following scenario just to make sure?
1. Declare `enable_data_masking = true` and do not declare `mask_all_occurrences`. It should work.
2. Declare `mask_all_occurrences = true` and do not declare `enable_data_masking`. It should complain.

Reply:
Great! Thanks!